### PR TITLE
Adding ChatClient with transport fallback and reconnects

### PR DESCRIPTION
**Pull request by moozzyk**

### CODE REVIEW

File: ChatSample/ChatClient/Program.cs [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+
+namespace ChatClient
+{
+    class Program
+    {
+        static async Task Main(string[] args)
+        {
+            try
+            {
+                await new ChatClient("test").Run();
```
Comment (moozzyk) [inline ChatSample/ChatClient/Program.cs]: The name was supposed to be read from the console...

File: ChatSample/ChatClient/ChatClient.cs [hunk 2]
```diff
@@ -0,0 +1,136 @@
...
+            }
+            catch (Exception ex)
+            {
+                Console.WriteLine($"Message not sent due to {ex.Message}");
+            }
+        }
+
+        private async Task<HubConnection> StartWithFallback()
+        {
+            for (var transport = TransportType.WebSockets;
```
Comment (davidfowl) [inline ChatSample/ChatClient/ChatClient.cs]: This loop looks pretty absurd 😄 .

File: ChatSample/ChatClient/ChatClient.cs [hunk 3]
```diff
@@ -0,0 +1,136 @@
...
+                a.Cancel = false;
+                _lifetimeCts.Cancel();
+            };
+
+            var exitTask = Task.Delay(-1, _lifetimeCts.Token);
+
+            while (!_lifetimeCts.IsCancellationRequested)
+            {
+                Console.Write("> ");
+                var result = await Task.WhenAny(Task.Run(() => Console.ReadLine()), exitTask);
```
Comment (davidfowl) [inline ChatSample/ChatClient/ChatClient.cs]: This code can be cleaner:

```C#
var result = await Task.WhenAny(Task.Run(() => Console.ReadLine()), exitTask);

if (result is Task<string> task)
{
     var line = await task;

    // line can be null when Ctrl-C is pressed
    if (line != null)
    {
        await SendMessage(line);
    }
}
else
{
    break;
}
```

Comment (rdcm) [reply ChatSample/ChatClient/ChatClient.cs]: more cleaner :)
```
var result = await Task.WhenAny(Task.Run(() => Console.ReadLine()), exitTask);
if (!(result is Task<string> task))
{
   break;
}

var line = await task;
if (line != null)
{
   await SendMessage(line);
}
```

**Comment by davidfowl** [comment]
We should do this over since we now have fallback and restartability

**Pull request closed without merging**